src/DateTimeParser.h

```diff
@@ -311,7 +311,10 @@ class DateTimeParser {
 
 private:
   int hour() {
-    return hour_ + (amPm_ == 1 ? 12 : 0);
+    if (hour_ == 12 && amPm_ == 0) {
```

Comment:
I feel like this might be simpler if you branched on `hour_ == 12` so you could also simplify the ternary if below.